lib/rust/ensogl/core/src/display/scene/layer.rs

```diff
@@ -624,12 +624,19 @@ impl LayerModel {
         self.sublayers.borrow().all()
     }
 
-    fn add_sublayer(&self, layer: &Layer) {
+    /// Attach a `layer` as a sublayer.
+    pub fn add_sublayer(&self, layer: &Layer) {
         let ix = self.sublayers.borrow_mut().layers.insert(layer.downgrade());
         self.sublayers.borrow_mut().layer_placement.insert(layer.id(), ix);
         layer.add_parent(&self.sublayers);
     }
 
+    /// Remove previously attached sublayer.
+    pub fn remove_sublayer(&self, layer: &Layer) {
+        self.sublayers.borrow_mut().remove(layer.id());
+        layer.remove_parent(&self.sublayers);
+    }
+
```

Comment:
Is this correctly updating the per-layer shapes registry? Probably it does not have to update it, but please, double check that this operation correctly hides elements on the stage.

Reply:
I checked it using a modified demo scene - it works as intended. The implementation is based on `remove_all_sublayers()` method.

Reply:
Sure. At the end of `remove_all_sublayers` you can see the following call:
```rust
mem::take(&mut *self.sublayers.model.borrow_mut());
```
Essentially the `sublayers.model` is replaced with `SublayersModel::default()` here. If we look at `SublayersModel`, it is a struct of two fields:
```rust
pub struct SublayersModel {
     layers:          OptVec<WeakLayer>,
     layer_placement: HashMap<LayerId, usize>,
}
```
`remove_all_sublayers` just clears these collections (`OptVec` and `HashMap`), replacing them with a fresh `Default::default()` results. But in `remove_sublayer` we only need to remove info about a specific sublayer. We can check how `add_sublayer` is implemented:
```rust
let ix = self.sublayers.borrow_mut().layers.insert(layer.downgrade());
self.sublayers.borrow_mut().layer_placement.insert(layer.id(), ix);
// ...
```

So it modifies both fields. To undo these changes, we simply need to remove a `WeakLayer` from `layers` and an entry from `layer_placement`. That's precisely how `remove_sublayer` was implemented.

Reply:
Perfect, add it as a short comment (shorter version) somewhere in the docs – either to docs of these functions or somewhere in one of their bodies.